### PR TITLE
Set d0i3 register on MTL

### PR DESCRIPTION
**Pull request by RanderWang**
All cavs platform use the same d0i3 register offset but MTL doesn't. This PR addresses this issue by introducing d0i3_offset in chip_info

### CODE REVIEW

File: sound/soc/sof/ops.h [hunk 1]
```diff
@@ -324,6 +324,27 @@ static inline void snd_sof_dsp_write64(struct snd_sof_dev *sdev, u32 bar,
...
+}
+
+static inline u8 snd_sof_dsp_readb(struct snd_sof_dev *sdev, u32 bar,
+				   u32 offset)
+{
+	if (sof_ops(sdev)->readb)
+		return sof_ops(sdev)->readb(sdev, sdev->bar[bar] + offset);
+
+	dev_err(sdev->dev, "error: %s not defined\n", __func__);
+	return (char)-EOPNOTSUPP;
```
Comment (ujfalusi) [inline sound/soc/sof/ops.h:345]: The caller will have hard time to figure out that the u8 casted error code is a value or error, Just return 0?
Yes, we shall fix this for the other variants as well.

**Review by ujfalusi — APPROVED** [review]

File: sound/soc/sof/intel/mtl.h [hunk 2]
```diff
@@ -21,6 +21,8 @@
 #define MTL_IRQ_INTEN_L_SOUNDWIRE_MASK	BIT(6)
 #define MTL_HFINTIPPTR_PTR_MASK		GENMASK(20, 0)
 
+#define MTL_HDA_VS_D0I3C	0x1D4A
```
Comment (ujfalusi) [inline sound/soc/sof/intel/mtl.h:24]: One more tab to align it?

File: sound/soc/sof/intel/hda-dsp.c [hunk 3]
```diff
@@ -378,27 +381,31 @@ static int hda_dsp_send_pm_gate_ipc(struct snd_sof_dev *sdev, u32 flags)
...
 	ret = hda_dsp_wait_d0i3c_done(sdev);
 	if (ret < 0) {
-		dev_err(bus->dev, "CIP timeout before D0I3C update!\n");
+		dev_err(sdev->dev, "CIP timeout before D0I3C update!\n");
 		return ret;
 	}
 
 	/* Update D0I3C register */
-	snd_hdac_chip_updateb(bus, VS_D0I3C, SOF_HDA_VS_D0I3C_I3, value);
+	snd_sof_dsp_writeb(sdev, HDA_DSP_HDA_BAR, chip->d0i3_offset, value);
```
Comment (ujfalusi) [inline sound/soc/sof/intel/hda-dsp.c:398]: updateb is not equivalent to writeb!

**Review by ujfalusi — CHANGES_REQUESTED** [review]

**Comment by plbossart** [comment]
@RanderWang it seems we both worked on the same topic, I added readb/writeb capabilities and modified the D0i3C yesterday :-) see https://github.com/thesofproject/linux/pull/3884

File: sound/soc/sof/intel/hda-dsp.c (hunk 3 above)
Comment (RanderWang) [reply sound/soc/sof/intel/hda-dsp.c:398]: o, thanks

**Comment by RanderWang** [comment]
> @RanderWang it seems we both worked on the same topic, I added readb/writeb capabilities and modified the D0i3C yesterday :-) see #3884

thanks! I will close it

**Comment by plbossart** [comment]
I'll extract to patches to a new PR so that you are not blocked @RanderWang

**Comment by RanderWang** [comment]
> I'll extract to patches to a new PR so that you are not blocked @RanderWang

Thanks! I can enjoy my vocation : )

**Pull request closed without merging**